File: Program's_Contributed_By_Contributors/Python_Programs/binarySearch.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2

        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1

    return -1  # If the target is not found


arr = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
target = 7

result = binary_search(arr, target)

if result != -1:
    print(f"Element found at index {result}")
else:
    print("Element not found")

File: Program's_Contributed_By_Contributors/C++ Programs/SearchingAlgorithms/binary_search.cpp
#include <iostream>
using namespace std;

int binarySearch(int array[], int x, int low, int high) {
  if (high >= low) {
    int mid = low + (high - low) / 2;

    // If element is at mid return it
    if (array[mid] == x)
      return mid;

    // Try to search on left half
    if (array[mid] > x)
      return binarySearch(array, x, low, mid - 1);

    // Try to search on right half
    return binarySearch(array, x, mid + 1, high);
  }

  return -1;
}

int main(void) {
	int n, x;
	cout<<"Enter the length of the array: ";
	cin>>n;
	int arr[n];
	cout<<"Enter the values you want to add to array: ";
	for(int i=0; i<n; i++){
		cin>>arr[i];
	}
	cout<<"Enter the value you want to find in the array: ";
	cin>>x;
  	int result = binarySearch(arr, x, 0, n - 1);
  	if (result == -1){
  	cout<<"Value not found in the array"<<endl;
  	}
  	else{
  	cout<<"Value found at index: "<<result<<endl;
  	}
}
